deployment/postgres-init-db/sql/tables/actions/action_definition.sql

```diff
@@ -0,0 +1,84 @@
+create table actions.action_definition (
+  id integer generated always as identity,
+
+  name text not null,
+  description text null,
+  parameter_schema jsonb not null default '{}'::jsonb,
+  settings_schema jsonb not null default '{}'::jsonb,
+  settings jsonb not null,
+
+  action_file_id integer not null,
+  workspace_id integer not null,
+
+  created_at timestamptz not null default now(),
+  owner text,
+  updated_at timestamptz not null default now(),
+  updated_by text,
+
+  constraint action_definition_synthetic_key
+    primary key (id),
+
+  foreign key (workspace_id)
+    references sequencing.workspace (id)
+    on delete set null,
```

Comment:
Bug: this foreign key will cause issues, as `workspace_id` is a `not null` column. I'm assuming that actions can only exist within the concept of a specific workspace. Therefore, this should be: 

```suggestion
  foreign key (workspace_id)
    references sequencing.workspace (id)
    on delete cascade,
```